pkg/controller/clone/common.go

```diff
@@ -0,0 +1,277 @@
+		ns = *pvc.Spec.DataSourceRef.Namespace
+	}
+
+	obj := &cdiv1.VolumeCloneSource{}
+	exists, err := getResource(ctx, c, ns, pvc.Spec.DataSourceRef.Name, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if !exists {
+		return nil, nil
+	}
+
+	return obj, nil
+}
+
+// AddCommonLabels adds common labels to a resource
+func AddCommonLabels(obj metav1.Object) {
+	if obj.GetLabels() == nil {
+		obj.SetLabels(make(map[string]string))
+	}
+	obj.GetLabels()[common.CDILabelKey] = common.CDILabelValue
+}
+
+// AddCommonClaimLabels adds common labels to a pvc
+func AddCommonClaimLabels(pvc *corev1.PersistentVolumeClaim) {
+	AddCommonLabels(pvc)
+	if util.ResolveVolumeMode(pvc.Spec.VolumeMode) == corev1.PersistentVolumeFilesystem {
+		pvc.GetLabels()[common.KubePersistentVolumeFillingUpSuppressLabelKey] = common.KubePersistentVolumeFillingUpSuppressLabelValue
+	}
+}
+
+// AddOwnershipLabel adds owner label
+func AddOwnershipLabel(label string, obj, owner metav1.Object) {
+	if obj.GetLabels() == nil {
+		obj.SetLabels(make(map[string]string))
+	}
+	obj.GetLabels()[label] = string(owner.GetUID())
+}
+
+// IsSourceClaimReady checks that PVC exists, is bound, and is not being used
+func IsSourceClaimReady(ctx context.Context, c client.Client, namespace, name string) (bool, error) {
+	claim := &corev1.PersistentVolumeClaim{}
+	exists, err := getResource(ctx, c, namespace, name, claim)
+	if err != nil {
+		return false, err
+	}
+
+	if !exists {
+		return false, nil
+	}
+
+	if claim.Status.Phase != corev1.ClaimBound {
+		return false, nil
+	}
+
+	pods, err := cc.GetPodsUsingPVCs(ctx, c, namespace, sets.New(name), true)
+	if err != nil {
+		return false, err
+	}
+
+	if len(pods) > 0 {
+		return false, nil
+	}
+
+	return cdiv1.IsPopulated(claim, dataVolumeGetter(ctx, c))
+}
+
+// GetGlobalCloneStrategyOverride returns the global clone strategy override
+func GetGlobalCloneStrategyOverride(ctx context.Context, c client.Client) (*cdiv1.CDICloneStrategy, error) {
+	cr, err := cc.GetActiveCDI(ctx, c)
+	if err != nil {
+		return nil, err
+	}
+
+	if cr == nil {
+		return nil, fmt.Errorf("no active CDI")
+	}
+
+	if cr.Spec.CloneStrategyOverride == nil {
+		return nil, nil
+	}
+
+	return cr.Spec.CloneStrategyOverride, nil
+}
+
+// GetStorageClassForClaim returns the storageclass for a PVC
+func GetStorageClassForClaim(ctx context.Context, c client.Client, pvc *corev1.PersistentVolumeClaim) (*storagev1.StorageClass, error) {
+	if pvc.Spec.StorageClassName == nil || *pvc.Spec.StorageClassName == "" {
+		return nil, nil
+	}
+
+	sc := &storagev1.StorageClass{}
+	exists, err := getResource(ctx, c, "", *pvc.Spec.StorageClassName, sc)
+	if err != nil {
+		return nil, err
+	}
+
+	if exists {
+		return sc, nil
+	}
+
+	return nil, nil
+}
+
+// MustGetStorageClassForClaim returns the storageclass for a PVC
+func MustGetStorageClassForClaim(ctx context.Context, c client.Client, pvc *corev1.PersistentVolumeClaim) (*storagev1.StorageClass, error) {
```

Comment:
I think this function name is a little bit awkward. I also think it doesn't make that much sense to have it since we still need to check its error return (we are only saving two checks in `planner.go`).